src/cdk/tree/tree.ts

```diff
@@ -1175,8 +1175,14 @@ export class CdkTreeNode<T, K = T> implements OnDestroy, OnInit, TreeKeyManagerI
 
   /** Determines if the tree node is expandable. */
   _isExpandable(): boolean {
-    if (typeof this._tree.treeControl?.isExpandable === 'function') {
-      return this._tree.treeControl.isExpandable(this._data);
+    if (this._tree.treeControl) {
+      if (typeof this._tree.treeControl?.isExpandable === 'function') {
+        return this._tree.treeControl.isExpandable(this._data);
+      }
+
+      // For compatibility with trees created using TreeControl before we added
+      // CdkTreeNode#isExpandable.
+      return true;
```

Comment:
it's not clear to me if backwards compatibility is desired in this case given that the original behaviour is incorrect from an a11y standpoint

Reply:
Angular components is required to follow semantic versioning. When releasing this in a minor version, we're required not to break apps that were made before before introducing `CdkTreeNode#isExpandable`. 

I know that the a11y is incorrect, but this is not causing an a11y regression. It means that consumers will get the correct behavior when moving off the the deprecated Tree Control. I know this isn't particularly satisfying, but we have to follow Angular's semantic versioning policy.

If it were for a major version, then we would have a bit more flexibility, but doing this in a minor will allow us to release this in a timely manor. There are a lot of a11y improvements in this PR, and I think it would be good to do what we can feasibly land.

I'm definitely open to suggestions if there is another way we can do this that is backwards compatible for Trees made before cdk-tree-revamp.

Reply:
We should document this then, I can make some edits to the readmes to that effect.